### DISCUSSION TITLE
Debian release

### DISCUSSION THREAD

**Discussion by bodograumann** [br3ndonland/inboard#80] [General]
Just wanted to note, that since v0.50 the base image is now debian bookworm instead of debian bullseye.
Might be worth a mention in the release notes.
For us this required a change, because we where installing software from bullseye-backports.

**Comment by br3ndonland** [comment]
Thanks for pointing out the Debian version update. I added a note to the [changelog entry for inboard 0.51.0](https://inboard.bws.bio/changelog#0510-2023-07-09). I have seen [a number of issues reported with Debian bookworm](https://github.com/docker-library/python/issues?q=bookworm). Changes to the underlying system libraries sometimes result in these unforeseen breaking changes.

Would it be helpful to have inboard manage the Debian version? Maybe like this:

- inboard could pull Docker images with a specific Debian version. For example, rather than pulling `FROM python:3.11` or `FROM python:3.11-slim` for Python 3.11 on Debian, inboard could pull `FROM python:3.11-bookworm` or `FROM python:3.11-slim-bookworm`.
- inboard would then do a minor release when updating Debian versions, like from bookworm to the next version.

As you point out, inboard's Docker images were automatically updated to bookworm when Docker updated their Python images, so I don't think we should revert back to bullseye at this point, but it could help to have inboard manage the Debian version going forward.

inboard could do the same thing with Alpine, but Alpine has been relatively stable so it might not be necessary.

**Reply by bodograumann** [reply]
That sounds like a good idea. With the current setup it is too easy to overlook the changes introduced by the release.

**Comment by br3ndonland** [comment]
Starting with version 0.67, inboard will use a specific Debian version when building Docker images, ensuring that the version does not change unexpectedly. The current Debian version is still Debian 12 ("bookworm"). The next Debian version, Debian 13 ("trixie") does not have a release date yet, but inboard will update to trixie when it is stable and will provide a new release after the update.
